### DISCUSSION TITLE
creating PR using Github REST API

### DISCUSSION THREAD

**Discussion by parteekcoder** [community/community#56853] [Programming Help]
### Body

Hi everyone,

In my project, I want to create a pull request from the repo to its upstream repo using GIthub REST API. I read the github docs and it states that to do that we need the write access to the upstream repo as well but why is that? While creating PR using Github we don't need the write access to the upstream repo.

Can someone help me to create PR without having access to upstream using Github REST API?

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by Venkatesh-KCET** [comment]
```python
from github import Github

def create_pull_request(repo_owner, repo_name, base_branch, head_branch, pr_title, pr_body, access_token):
    g = Github(access_token)
    repo = g.get_repo(f"{repo_owner}/{repo_name}")
    base = repo.get_branch(base_branch)
    head = repo.get_branch(head_branch)
    pr = repo.create_pull(
        title=pr_title,
        body=pr_body,
        base=base.name,
        head=head.name
    )
    print(f"Pull request created successfully! PR number: {pr.number}")

# Example usage
repo_owner = "your_username"
repo_name = "your_repository"
base_branch = "main"
head_branch = "feature_branch"
pr_title = "My Pull Request"
pr_body = "This is the description of my pull request."
access_token = "your_personal_access_token"

create_pull_request(repo_owner, repo_name, base_branch, head_branch, pr_title, pr_body, access_token)
```

Make sure to replace `"your_username"`, `"your_repository"`, `"feature_branch"`, `"My Pull Request"`, `"This is the description of my pull request."`, and `"your_personal_access_token"` with your actual values.

The `pygithub` module provides a more convenient and Pythonic way to interact with the GitHub API, abstracting away some of the low-level HTTP requests and responses.
